%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%Preamble
\documentclass{article}

%Dependencies
\usepackage[left]{lineno}
\usepackage{titlesec}
\usepackage{xcolor}

\newcommand\hl[1]{%
  \bgroup
  \hskip0pt\color{blue!80!black}%
  #1%
  \egroup
}
\usepackage{ogonek}
\usepackage{float}
\usepackage{times}
\usepackage{amsmath}
\usepackage{old-arrows}
\usepackage{enumitem}
\usepackage{wasysym}
\usepackage[titletoc,title]{appendix}

% Other Packages
%\usepackage{times}
\RequirePackage{fullpage}
\linespread{1.5}
\RequirePackage[colorlinks=true, allcolors=black, linktoc=page]{hyperref}
\RequirePackage[english]{babel}
\RequirePackage{amsmath,amsfonts,amssymb}
\usepackage{mathptmx}
%\RequirePackage[sc]{mathpazo}
\RequirePackage[T1]{fontenc}
\RequirePackage{url}
\usepackage{tabu}

% Bibliography
%\usepackage[authoryear,sectionbib,sort]{natbib}
\usepackage{natbib} \bibpunct{(}{)}{;}{author-year}{}{,}
\bibliographystyle{evolution}
\addto{\captionsenglish}{\renewcommand{\refname}{References}}
\setlength{\bibsep}{0.0pt}

% Graphics package
\usepackage{graphicx}
\graphicspath{{../output/figures/}.pdf}

% Change Appendix Numbering
%\renewcommand\thesection{Appendix \Alph{section}}
\addcontentsline{toc,page}{section}{Appendix~\ref{section}}
\renewcommand\thesubsection{\Alph{section}.\arabic{subsection}}

% New commands: fonts
\def\mbf#1{\mathbf{#1}}

%\newcommand{\code}{\fontfamily{pcr}\selectfont}
%\newcommand*\chem[1]{\ensuremath{\mathrm{#1}}}
\newcommand\numberthis{\addtocounter{equation}{1}\tag{\theequation}}
%\titleformat{\subsubsection}[runin]{\bfseries\itshape}{\thesubsubsection.}{0.5em}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Supporting Information (Appendices A--\hl{E}) for: Consequences of recessive deleterious genetic variation for the evolution of inversions suppressing recombination between sex chromosomes. \textit{Evolution}}

%\author{Colin Olito$^{\ast}$, Bengt Hansson, Suvi Ponnikas$^{1}$, and Jessica K.~Abbott}
\date{\today}

\begin{document}
\maketitle

%\noindent{} Department of Biology, Lund University, Lund 223 62, Sweden.

%\noindent{} $^1$ Current Address: \hl{Suvi's Current Address}.

%\noindent{} $^{\ast}$ Corresponding author e-mail: \url{colin.olito@gmail.com}
\bigskip

\noindent{} Additional computer code and supplementary info available at: \url{https://github.com/colin-olito?tab=repositories}

\bigskip

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Running Header
%\pagestyle{fancyplain}
%\makeatother
%\lhead{\textit{Supplement to Olito. Linkage and SA polymorphism in hermaphrodites. \textit{Evolution}.\\}}
%%\rhead{\textit{Sex antagonistic selection on phenology}}
%\renewcommand{\headrulewidth}{0pt}
%\renewcommand{\footrulewidth}{0pt}
%\addtolength{\headheight}{12pt}

\tableofcontents

\newpage

\begin{appendices}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \section{Heterozygote advantage and 'sheltering' hypotheses} \label{Sheltering-App}
 \renewcommand{\theequation}{A\arabic{equation}}
 \setcounter{equation}{0}
 \renewcommand{\thefigure}{A\arabic{figure}}
 \setcounter{figure}{0}

The idea that linkage to the sex-determining loci can cause a reduction in homozygous expression of deleterious alleles at nearby loci has appeared several times in the literature. Unfortunately, the various verbal and mathematical models can be confusing because they have been formulated with different biological scenarios in mind (e.g., proto sex, neo-sex, and mating-type chromosomes), and have cited different theoretical studies as support. Below, we briefly summarize and quote relevant passages from key articles that have proposed or referenced this idea as either a verbal or mathematical model. We have added emphasis to particularly important passages in the quotations.


\subsection*{Ironside (2010)}

To our knowledge, \citet{Ironside2010} was the first to suggest that linkage to the dominant ({\itshape i.e.}, permanently heterozygous) sex-determining allele could cause a reduction in homozygous expression of deleterious mutations, and thereby indirect selection for suppressed recombination between sex chromosomes. Ironside's key contribution was to point out that the earlier models of \citet{CharlesworthWall1999}, which involved the formation of neo-sex chromosomes by nonreciprocal translocations in inbreeding populations, might be applicable to the evolution of suppressed recombination between sex chromosomes. In the subsection titled "Alternative hypotheses for the evolution of non-recombining regions", \citet{Ironside2010} states:
	\begin{quote} 
		"Three alternative mechanisms are proposed to explain the spread of non-recombining regions on sex chromosomes ... The second is that suppression of recombination is selected because it prevents homozygosity of deleterious recessive genes, particularly in populations with inbreeding [33]."\\
		\footnotesize{References: [33] \citet{CharlesworthWall1999}}
	\end{quote}

\noindent A second passage several paragraphs later elaborates on the hypothesis:

	\begin{quote}
		"The hypothesis that chromosomal rearrangements spread through selection to prevent homozygosity of deleterious recessive genes at multiple loci was proposed by Charlesworth and Wall [33]. Their models demonstrate that, in populations with moderate levels of inbreeding, selection to maintain heterozygosity at two loci can favor the spread of neo-sex chromosomes generated by centric fusions or reciprocal translocations."
	\end{quote}

\noindent Overall, it is difficult to know exactly what \citet{Ironside2010} was proposing. The relevant passages are not explicit about whether the hypothetical chromosomal rearrangement is selectively favored because it captures wild-type alleles, deleterious alleles, or a particular combination of both. Also, although the second passage clearly refers to neo-sex chromosome formation, the hypothesis was brought up in a discussion about the expansion of non-recombining regions by inversions surrounding the sex-determining loci on proto sex chromosomes. For example, Fig.~1 and Fig.~2 of \citet{Ironside2010} very clearly depict inversions expanding the non-recombining region around a sex-determining locus, not neo-sex chromosome formation. It is not quite clear whether the mechanism is implied to work for the evolution of recombination suppression on proto sex chromosomes. 


\subsection*{Ponnikas et al.~(2018)}

In their review of theories for the suppression of recombination between sex chromosomes (and empirical evidence supporting them), \citet{Ponnikas2018} refer to hypotheses involving "Heterozygote Advantage". The authors describe a very similar process to \citet{Ironside2010} and cite \citet{CharlesworthWall1999} and an older paper about translocations and mutational heterosis in completely selfing species \citep{deWaalMaleFijtCharlesworth1979}:

	\begin{quote}
		Heterozygosity increases fitness by concealing recessive deleterious mutations and causing overdominance at functional loci [28]. Thus, it can be hypothesised that heterozygote advantage around a sex-determining gene in the heterogametic sex can favour recombination suppression (because less recombination means a larger heterozygote region). A challenge is however to explain how the sex-determining gene can be associated with a sufficient amount of inbreeding load (deleterious recessives) for the heterogametic sex to overcome the fitness costs of establishing genetic sex determination in the first place (induced by the skew in sex ratio, mentioned earlier; cf. [27]). A possible scenario is that the sex-determining gene is part of a larger rearrangement (reciprocal translocation or inversion) that captures a suite of loci carrying deleterious mutations, which then becomes fixed for heterozygosity in the heterogametic sex, which therefore experiences higher fitness due to heterozygote advantage. This implies that heterozygote advantage in principle is a possible agent of sex chromosome formation in evolutionary scenarios where inbreeding avoidance is favourable (Table 1).\\
		\footnotesize{References: [27] \citet{deWaalMaleFijtCharlesworth1979}, [28] \citet{CharlesworthWall1999}}
	\end{quote}

\noindent \citet{Ponnikas2018} are explicit in describing the linkage of loci with segregating recessive deleterious mutations to the sex-chromosomes but are ambiguous about which alleles ultimately become Y-linked. As in \citet{Ironside2010}, the authors clearly note that the models of \citet{CharlesworthWall1999} refer to neo-sex chromosome evolution (and even provide a detailed schematic figure of neo-sex chromosome formation). However, it remains unclear whether they view this as a completely different process from the expansion of a non-recombining region around sex-determining loci. Their citation of \citet{deWaalMaleFijtCharlesworth1979} in this context also introduces some confusion because this paper was motivated by the rather extreme example of heterozygosity of translocation complexes in the autogamous flowering plant genus {\itshape Oenothera}, and several key assumptions of the models they present are peculiar to obligately selfing hermaphrodites. 


\subsection*{Charlesworth \& Wall (1999)}

 This important paper described the invasion of centric fusions or translocations causing a single autosomal locus with heterozygote advantage to become completely linked to either the Y or X chromosome in a partially inbreeding population (partial full-sib mating). Here, we give a brief overview but refer readers to the original article for further details. The authors made two statements suggesting that their results might be relevant to scenarios involving selected loci under deleterious mutation pressure. In the last paragraph of the introduction, the authors explain that their choice of model was a practical one:

	\begin{quote}
		"There is therefore a need to investigate quantitatively the question of the nature of selection for neo-X or neo-Y chromosomes in inbreeding populations. The causes of inbreeding depression and heterosis are still a matter for debate, but it seems likely that both deleterious mutations and alleles maintained by heterozygote advantage play a role, with the former probably being more important (Crow 1993; Charlesworth 1998). \bf{It is, however, much simpler to model the case of a single locus with heterozygote advantage rather than a multi-locus model of mutation and selection, and this is the subject of the present paper}. The results confirm that there is, indeed, often selection for a fusion or translocation between a sex chromosome and an autosome in a partially inbreeding species when there is heterozygote advantage."
	\end{quote}

\noindent In the final sentences of the Discussion, the authors make the following extremely cautious conjecture regarding the relevance of their findings to the case of deleterious mutations:

	\begin{quote}
		"Second, we have assumed that heterosis is caused by a single locus with heterozygote advantage. \bf{Since homozygosity for a chromosome segment in an inbreeding population is likely to be associated with reduced fitness in much the same way as homozygosity for an allele at a locus with heterozygote advantage, the present study probably provides a rough guide to what is likely to be true in the case of mutational heterosis (Charlesworth 1991). A detailed investigation of this case by computer simulations is needed to verify this conjecture}."
	\end{quote}

\noindent Subsequent references to the models of \citet{CharlesworthWall1999} by other authors to explain either neo-sex chromosome formation {\itshape \underline or} recombination suppression between proto sex chromosomes have introduced a couple points of confusion. First, and most importantly, because the original models used a single locus under overdominant selection, all subsequent references to them as support for the idea that linkage to the sex-determining locus can reduce homozygote expression of deleterious alleles is based on the conjecture rather than model predictions. Second, despite this issue of referencing, the models of \citet{CharlesworthWall1999} are still relevant to the process of recombination suppression between proto sex chromosomes. Specifically, they become mathematically equivalent to a model of the expansion of non-recombining region surrounding the sex-determining loci to capture a selected locus under heterozygote advantage, provided there is free recombination between the sex-determining and selected loci on the wild-type karyotype. Despite the confusion surrounding Charlesworth \& Wall's conjecture, the scenarios of neo-sex chromosome formation and expansion of the non-recombining region around the sex-determining loci by an inversion (or other recombination modifier) are conceptually and mathematically linked. Unfortunately, neither of the above review articled made this connection, or the distinction betwen the scenarios, explicit. 

Overall, it seems clear that Charlesworth \& Wall's conjecture regarding the applicability of their model results to the case of mutational heterosis deserves closer scrutiny. In \ref{CW1999-App}, we revisit the models of \citet{CharlesworthWall1999} and expand them to address the case of an inversion linking multiple selected loci under deleterious mutation pressure to the dominant male-determining allele on a proto sex chromosome. Our results indicate that the conjecture, upon which the later reviews and citing articles have based their argumentation, does not appear to work.


\subsection*{Branco et al. (2017)}

\citet{Branco2017} propose a very similar hypothesis to those of \citet{Ironside2010} and \citet{Ponnikas2018}, but approach the idea from a different biological scenario: the evolution of recombination suppression on haploid mating-type chromosomes in some fungi. However, \citet{Branco2017} are explicit in stating that deleterious alleles must become linked to mating-type or sex-determining loci. They propose the following hypothesis in their Supplementary Material:

	\begin{quote}
		"Another evolutionary explanation for suppressed recombination on fungal mating-type chromosomes, {\bf that could also apply to sex chromosomes}, involves linkage of deleterious alleles to the mating-type loci, favoring permanent sheltering in an heterozygous state, as has been theoretically modeled (20, 21). This may arise specifically in sex and mating-type chromosomes if recombination frequency gradually decreases from the non-recombining into the PAR, so that partial linkage (linkage disequilibrium) to mating-type or sex-determining genes makes selection against recessive deleterious alleles less efficient. This would allow deleterious alleles to increase in frequency in these PAR edges at the margin of the non-recombining region. Rare recombination events would then generate individuals homozygous for deleterious alleles and could therefore be selected against. Complete linkage of these PAR margins in disequilibrium with mating type may thus be favorable and selected for (Fig. S1A). Permanent sheltering would thus be more easily achieved than purging if recombination is rare at the PAR margins. Further theoretical models are needed to explore the general conditions under which such a mechanism can generate evolutionary strata."\\
		\footnotesize{References: (20) Antonovics \& Abrams (2004); (21) Johnson et al.~ (2001).}
	\end{quote}

\noindent Branco et al.(2017)'s verbal model also introduces confusion because the models of mating-type chromosomes they cite involve a variety of different assumptions that, while reasonable for some fungi species with haploid sex-determination, do not hold for species with diploid genetic sex-determination. Moreover, this verbal model introduces a new complicating factor that was not present in the models of \citet{CharlesworthWall1999}: genetic linkage between the mating-/sex-determining loci and load loci prior to the evolution of arrested recombination between them. In their Fig. S1A, they explicitly depict deleterious mutations becoming linked to mating-type alleles.

In \ref{Branco-App}, we address the potential for linkage disequilibrium between the sex-determining and selected loci to influence the invasion of a recombination modifier suppressing recombination between them.


\subsection*{Jay et al.~(2021)}

A recent preprint published on bior$\chi$iv titled 'A deleterious mutation-sheltering theory for the evolution of sex chromosomes and supergenes' presented the most explicit hypothesis to date involving the evolution of recombination suppression beteen sex chromosomes due to 'sheltering' of recessive deleterious alleles on the Y chromosome (access the article \href{https://www.biorxiv.org/content/10.1101/2021.05.17.444504v1.article-info}{\color{blue} here}). In this manuscript, the authors explicitly propose that inversions expanding the sex-linked region on a Y chromosome that capture strongly recessive deleterious alleles will enjoy a permanent selective advantage because those deleterious alleles will be permanently sheltered as heterozygotes on the Y. They state the hypothesis verbally in the paper and attempt to formalize it as a mathematical model. Here is a representative quotation from the introduction:

	\begin{quote}
		Now, consider an inversion that, by chance, is in perfect linkage with a permanently heterozygous allele, such as the male-determining allele in a XY system. \bf{If this sex-linked inversion captures fewer deleterious variants than the population average, it should increase in frequency without ever suffering the deleterious consequences of having its load expressed. The recessive deleterious mutations captured by the sex-linked inversion are indeed fully linked to the permanently heterozygous, male-determining allele, and will, therefore, never occur as homozygotes. They are therefore sheltered from selection}. Hence, such an inversion would be expected to spread, becoming perfectly associated with the male-determining allele, resulting in the suppression of recombination between the X and Y chromosomes in the region covered by the inversion.
	\end{quote}

\noindent The mathematical model presented in the article formalizes the above verbal model by assuming that descendent copies of the inversion do not accumulate new mutations at sites where they initially captured a wild-type allele. Under this assumption, so long as the inversion captures less than the average number of deleterious alleles in the chromosomal segment it spans, it will be forever beneficial. As we detail in the present article, the time-dependent selection processes affecting inversions expanding the sex-linked region (SLR) are more complicated than this verbal model suggests. Though lightly-loaded inversions may be intially favoured, they eventually become deleterious as new mutations accumulate on descendent copies of the inversion. Initially loaded SLR-expanding inversions can fix during this window of time while they are still beneficial, but they do so despite carrying deleterious alleles, not because those deleterious alleles are sheltered from expression as homozygotes.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Development of the deterministic model} \label{DetermModel-App}
 \renewcommand{\theequation}{B\arabic{equation}}
 \setcounter{equation}{0}
 \renewcommand{\thefigure}{B\arabic{figure}}
 \setcounter{figure}{0}

To model the evolution of new SLR-expanding inversions, we need to derive recursions describing per-generation gene frequency changes for two categories of loci: ($1$) loci where the inversion intially captures a wild-type allele and ($2$) those where it captures a deleterious allele, within four different chromosome classes, ($1$) X's in ovules/eggs, ($2$) X's in pollen/sperm, ($3$) non-inverted Y's, and ($4$) inverted Y's. Under our assumption that the selected loci are unlinked all but the inverted Y chromosome class, we can derive these recursions from a simple 2-locus PAR model involving a sex-determining locus and second a selected locus under delterious mutation pressure. We then use the resulting recursions to formulate a model with multiple selected loci at linkage equilibrium. All of the same model assumptions noted in the main text still apply ({\itshape i.e.}, large population size, discrete generations, etc.)


\subsection{General 2-locus haplotype recursions}

Consider the following two-locus genetic system: one locus determines whether a chromosome is considered X or Y, with XX individuals being female, and XY individuals being male; and a second locus that may be linked to the sex-determining region and is subject to natural selection. At the selected locus, the wild-type allele, $A$, mutates to a deleterious variant, $a$, at a rate $\mu$ per chromosome (we ignore backmutation from $a \rightarrow A$). Recombination between the two loci occurs at a rate $r$ per meiosis. Generations are discrete, and the population size is assumed to be large enough that drift is negligible. The life cycle proceeds: mutation $\rightarrow$ selection $\rightarrow$ random mating.

There are three relevant female genotypes: $AA$, $Aa$, and $aa$, with frequencies denoted $x_1$, $x_2$, and $x_3$, and general fitness expressions at selection denoted $w_{f,1}$, $w_{f,2}$, $w_{f,3}$. However, there are eight relevant genotypes for males: $AA$, $AA^I$, $Aa$ (cis-), $Aa^I$ (cis-), $aA$ (trans-), $aA^I$ (trans-), $aa$, and $aa^I$, with frequencies $y_{1}$, $y^I_{1}$, $y_{2c}$, $y^I_{2c}$, $y_{2t}$, $y^I_{2t}$, $y_{3}$, and $y^I_{3}$, with fitness expressions $w_{m,1}$, $w_{m,2}$, $w_{m,3}$. Note that male heterozygote genotype labels indicate whether the $A$ allele is located on the X chromosome ($y_{2c}$), or on the Y chromosome ($y_{2t}$), and "$I$" superscripts denote inverted haplotypes. It is assumed that recombination is completely suppressed between inverted and non-inverted chromosomes. 

We denote the frequency of each of the relevant haplotypes as follows: 
\begin{align*}
	&X_{A,Ov}:~\text{the frequency of proto-X chromosomes carrying the {\itshape A} allele in ovules/eggs,}\\ 
	&X_{a,Ov}:~\text{the frequency of proto-X chromosomes carrying the {\itshape a} allele in ovules/eggs,}\\ 
	&X_{A,Sp}:~\text{the frequency of proto-X chromosomes carrying the {\itshape A} allele in pollen/sperm,}\\ 
	&X_{a,Sp}:~\text{the frequency of proto-X chromosomes carrying the {\itshape a} allele in pollen/sperm, and}\\ 
	&Y_A, Y_A^{I}, Y_a, Y_a^{I}:~\text{the frequency of inverted \& non-inverted proto-Y chromosomes}\\ &~~~~~~~~~~~~~~~~~~~~~~~~\text{carrying each allele in pollen/sperm.}
\end{align*}

\noindent The frequency of the three female genotypes after random mating are equal to:
\begin{align*}
		x_1 = & X_{A,Ov}  X_{A,Sp} \\
		x_2 = & X_{A,Ov}  X_{a,Sp} + X_{a,Ov} X_{A,Sp} \\
		x_3 = & X_{a,Ov}  X_{a,Sp} 
\end{align*}

\noindent and the frequency of the eight relevant male genotypes after random mating are:
\begin{align*}
		y_1      &= X_{A,Ov} Y_A \\
		y^I_1    &= X_{A,Ov} Y^I_A \\
		y_{2c}   &= X_{A,Ov} Y_a \\
		y^I_{2c} &= X_{A,Ov} Y^I_a \\
		y_{2t}   &= X_{a,Ov} Y_A \\
		y^I_{2t} &= X_{a,Ov} Y^I_A \\
		y_{3}    &= X_{a,Ov} Y_a \\
		y^I_{3}  &= X_{a,Ov} Y^I_a 
\end{align*}

\noindent The genotypic frequencies among females after mutation are:
\begin{align*}
	x^u_{1} &= x_1 (1 - 2 \mu) \\
	x^u_{2} &= x_2 (1 - \mu) + 2 x_1 \mu \\
	x^u_{3} &= x_3 + x_2 \mu \numberthis
\end{align*}

\noindent and among males:
\begin{align*}
	y^u_{1}      &= y1 (1 - 2 \mu) \\
	y^{I,u}_{1}  &= y^I_1 (1 - 2 \mu) \\
	y^u_{2c}     &= y_{2c} (1 - \mu) + y_1 \mu \\
	y^{I,u}_{2c} &= y^{I}_{2c} (1 - \mu) + y^I_1 \mu \\
	y^u_{2t}     &= y_{2t} (1 - \mu) + y_1 \mu \\
	y^{I,u}_{2t} &= y^I_{2t} (1 - \mu) + y^I_1 \mu \\
	y^u_{3}      &= y_3 + (y_{2c} + y_{2t}) \mu; \\
	y^{I,u}_{3}  &= y^I_3 + (y^I_{2c} + y^I_{2t}) \mu  \numberthis
\end{align*}

\noindent The haplotype frequencies among female gametes (ovules/eggs) after selection and meiosis are:
\begin{align*}
	X_{A,Ov}^{\prime} &= \Big( x^u_{1} w_{f,1} + \frac{x^u_{2} w_{f,2} }{2} \Big) \Big/ \overline{w}_f \\
	X_{a,Ov}^{\prime} &= \Big( x^u_{3} w_{f,3} + \frac{x^u_{2} w_{f,2} }{2} \Big) \Big/ \overline{w}_f  \numberthis
\end{align*}

\noindent where $\overline{w}_f = x^u_{1} w_{f,1} + x^u_{2} w_{f,2} + x^u_{3} w_{f,3}$. The corresponding haplotype frequencies among pollen/sperm are:
\begin{align*}
	X^{\prime}_{A,Sp} &= \big(y^u_1 w_{m,1} + y^{I,u}_1 w_{m,1} + y^u_{2c} w_{m,2} (1 - r) + y^{I,u}_{2c} w_{m,2} + y^u_{2t} w_{m,2} r \big) \big/ \overline{w}_m \\
	X^{\prime}_{a,Sp} &= \big(y^u{2c} w_{m,2} r + y^u_{2t} w_{m,2} (1 - r) + y^{I,u}_{2t} w_{m,2} + y^u_3 w_{m,3} + y^{I,u}_3 w_{m,3} \big) \big/ \overline{w}_m \\
	Y^{\prime}_{A} &= \big( y^u_1 w_{m,1} + yu2c*w_{m,2} r + y^u_{2t} w_{m,2} (1 - r) \big) \big/ \overline{w}_m \\
	Y^{I \prime}_{A} &= \big( y^{I,u}_1 w_{m,1} + (y^{I,u}_{2t} w_{m,2}) \big) \big/ \overline{w}_m \\
	Y^{\prime}_{a} &= \big( y^u_{2c} w_{m,2} (1 - r) + y^u_{2t} w_{m,2} r + y^u_3 w_{m,3} \big) \big/ \overline{w}_m \\
	Y^{I \prime}_{a} &= \big( y^{I,u}_{2c} w_{m,2} + y^{I,u}_3 w_{m,3} \big) \big/ \overline{w}_m \numberthis
\end{align*}

\noindent where $\overline{w}_{m} = y^u_1 w_{m,1} + y^{I,u}_1 w_{m,1} + y^u_{2c} w_{m,2} + y^{I,u}_{2c} w_{m,2} + y^u_{2t} w_{m,2} + y^{I,u}_{2t} w_{m,2} + y^u_3 w_{m,3} + y^{I,u}_3 w_{m,3}$.

\bigskip

\noindent The following substitutions allow us to simplify things: $X_{A,Ov} = 1 - X_{a,Ov}$, $X_{A,Sp} = 1 - X_{a,Sp}$, $Y_{A} = 1 - Y_a - Y_A^I - Y_a^I$. Also, because we assume the selected locus is initially in linkage equilibrium with the sex-linked region, the rate of recombination per meiosis between the two loci will be $r = 1/2$. We can now describe the frequency dynamics with a system of $5$ simplified haplotype recursions: $X^{\prime}_{a,Ov}$, $X^{\prime}_{a,Sp}$, $Y^{\prime}_{a}$, $Y^{\prime I}_{A}$, and $Y^{\prime I}_{a}$.
\bigskip

Ultimately, we want to be able to express our recursions in terms of the frequency of deleterious alleles in each of the different chromosome classes: $X$ chromosomes in ovules/eggs, $X$ chromosomes in pollen/sperm, non-inverted $Y$ chromosomes, and inverted $Y$ chromosomes. To do this, we must transform our haplotype recursions onto a new coordinate system by introducing $3$ new variables: $Y_I$, the frequency of the inversion among $Y$ chromosomes; $q_Y$, the frequency of the deleterious $a$ allele at the selected locus among non-inverted $Y$ chromosomes; and $q^I_Y$, the frequency of the deleterious $a$ allele at the selected locus among inverted $Y$ chromosomes. We can then make the following substitutions: $Y_a = (1 - Y_I)q_Y$, $Y^I_a = Y^I q_Y$, and $Y^I_A = Y_I (1 - q_Y)$, and define three new recursions:
\begin{align*}
	q^{\prime}_Y &= Y^{\prime}_a / (Y^{\prime}_A + Y^{\prime}_a) \\
	q^{\prime}_I &= Y^{\prime I}_a / (Y^{\prime I}_A + Y^{\prime I}_a) \\
	Y^{\prime}_I &= (Y^{\prime I}_A + Y^{\prime I}_a) / (Y^{\prime}_A + Y^{\prime}_a + Y^{\prime I}_A + Y^{\prime I}_a)\numberthis
\end{align*}

\noindent For consistency of notation, we relabel $X_{a,Ov} = q_{X_f}$ and $X_{a,Sp} = q_{X_m}$. We can now describe the frequency dynamics with a the following $5$ recursions: $q^{\prime}_{X_f}$, $q^{\prime}_{X_m}$, $q^{\prime}_Y$, $q^{\prime}_I$, and $Y^{\prime I}$. We develop the multilocus recursion for $Y^{\prime I}$ below in \ref{subsec:multilocYI}.


\subsection{Recursions when inversion captures either a wild-type or deleterious allele}

A new single-copy inversion mutation spanning both the sex-determining and selected loci will capture either a wild-type ($A$) or deleterious ($a$) allele at the selected locus. As explained in the main text, the allele frequency dynamics among each of the chromosome classes will differ depending on which allele the inversion captures. In the case where a new (rare) inversion captures a wild-type allele, the above recursions can be used to describe the resulting per-generation allele frequency changes in each chromosome class. We can model the case where a new inversion captures the deleterious $a$ allele by substituting $Y^I_A = 0$ into the above recursion system. In this case all descendent copies of the inversion will carry that allele ({\itshape i.e.}, $q_I = 1$ for all $t$ generations in the future). 

The full two-locus recursions for $q^{\prime}_{X_f}$, $q^{\prime}_{X_m}$, $q^{\prime}_Y$, and $q^{\prime}_I$ when the inversion captures either a wild-type or deleterious allele are presented in the accompanying Mathematica notebook file in the Online Supplementary Material. 


\subsection{Multilocus recursion for inversion frequency} \label{subsec:multilocYI}

To describe the frequency dynamics of an inversion that captures any number of unlinked selected loci, we have to make several simplifying assumptions (outlined in the main text). In brief, we assume the inversion captures $n$ loci, and that the selection parameters are constant across all captured loci ({\itshape i.e.,} $s_i = s$ and $h_i = h$). We can then categorize the loci captured by the inversion as {\itshape wt} and {\itshape del} depending on which allele is captured. The gene frequencies in each at each of the $n$ loci at time $t$ can be described using the following notation: $q^{del}_{X_f,t}$, $q^{del}_{X_m,t}$, $q^{del}_{Y,t}$, $q^{del}_{Y^I,t}$, and $q^{wt}_{X_f,t}$, $q^{wt}_{X_m,t}$, $q^{wt}_{Y,t}$, $q^{wt}_{Y^I,t}$, where $q$ refers to the deleterious allele frequency, and $del$ and $wt$ superscripts denote which allele was initially captured by the inversion at the $i^{th}$ locus. To simplify notation, we use the convention $p^{\cdot}_{\cdot,t} = 1 - q^{\cdot}_{\cdot,t}$. 

We can now define the following recursion for $Y^{\prime}_I$ that takes into account the fitness effects of the alleles it captures at all $n$ loci:
\begin{equation} \label{eq:YIprime-multiLoc}
	Y^I_{(t + 1)} = Y^I_t \Bigg[ \Big(1 - s \big(h p^{del}_{X_f,t} + q^{del}_{X_f,t} \big) \Big)^{r}\Big(1 - s \big(h (p^{wt}_{Y^I,t} q^{wt}_{X_f,t} + q^{wt}_{Y^I,t} p^{wt}_{X_f,t}) + q^{wt}_{Y^I,t} q^{wt}_{X_f,t} \big) \Big)^{n-r} \Bigg] \Bigg/ \overline{w}^Y
\end{equation}

\noindent where 
\begin{align*}\label{eq:wBarY-multiLoc}
	\overline{w}^Y = Y^I_t &\Bigg[ \Big(1 - s \big(h p^{del}_{X_f,t} + q^{del}_{X_f,t} \big) \Big)^{r}\Big(1 - s \big(h (p^{wt}_{Y^I,t} q^{wt}_{X_f,t} + q^{wt}_{Y^I,t} p^{wt}_{X_f,t}) + q^{wt}_{Y^I,t} q^{wt}_{X_f,t} \big) \Big)^{n-r} \Bigg] \times \\
	&(1 - Y^I_t) \left[\begin{array}{c}
											\bigg( 1 - s \Big( h \big( p^{del}_{X_f,t} q^{del}_{Y,t} +  q^{del}_{X_f,t} p^{del}_{Y,t} \big) + q^{del}_{X_f,t} q^{del}_{Y,t} \Big) \bigg)^r\\
											\bigg( 1 - s \Big( h \big( p^{wt}_{X_f,t} q^{wt}_{Y,t} +  q^{wt}_{X_f,t} p^{wt}_{Y,t} \big) + q^{wt}_{X_f,t} q^{wt}_{Y,t} \Big) \bigg)^{n-r}
											\end{array} \right] \numberthis
\end{align*}

\noindent Noting that $q^{del}_{I,{t+1}} = 1$. Equations (\ref{eq:YIprime-multiLoc}) and (\ref{eq:wBarY-multiLoc}) are presented in the main text as equations (1) and (2) in the main text.



%In contrast to our deterministic model results, our Wright-Fisher simulations allow the frequency of deleterious alleles at selected loci in X chromosomes and non-inverted Y chromosomes to drift independently (recall that we still constrain the frequency of deleterious alleles at {\itshape wt} loci on inverted Y chromosomes to follow deterministic dynamics). To correctly take the resulting differences in allele frequencies at each locus into account, we must modify the multilocus recursion for inversion frequency described in Eq(\ref{eq:YIprime-multiLoc}) and Eq(\ref{eq:wBarY-multiLoc}). Specifically, we must take the product over all $r$ del loci and $n â€“ r$ {\itshape wt} loci rather than exponentiating. This gives the following recursion for per-generation change in inversion frequencies, which we used in our Wright-Fisher simulations:

%\begin{equation} \label{eq:YIprime-multiLocProd}
%	Y^I_{(t + 1)} = Y^I_t  \Bigg[ \prod_{i=1}^{r}\Big(1 - s \big(h p^{del}_{X_{f,i},t} + q^{del}_{X_{f,i},t} \big) \Big) \prod_{j=1}^{n-r} \Big(1 - s \big(h (p^{wt}_{Y^I,t} q^{wt}_{X_{f,j},t} + q^{wt}_{Y^I,t} p^{wt}_{X_{f,j},t}) + q^{wt}_{Y^I,t} q^{wt}_{X_{f,j},t} \big) \Big) \Bigg] \Bigg/ \overline{w}^Y
%\end{equation}

%\noindent where 
%\begin{align*}\label{eq:wBarY-multiLocProd}
%	\overline{w}^Y = Y^I_{(t + 1)} = Y^I_t & \Bigg[ \prod_{i=1}^{r}\Big(1 - s \big(h p^{del}_{X_{f,i},t} + q^{del}_{X_{f,i},t} \big) \Big) \prod_{j=1}^{n-r} \Big(1 - s \big(h (p^{wt}_{Y^I,t} q^{wt}_{X_{f,j},t} + q^{wt}_{Y^I,t} p^{wt}_{X_{f,j},t}) + q^{wt}_{Y^I,t} q^{wt}_{X_{f,j},t} \big) \Big) \Bigg] \times \\
%	&(1 - Y^I_t) \left[\begin{array}{c}
%											\prod_{i=1}^{r} \bigg( 1 - s \Big( h \big( p^{del}_{X_{f,i},t} q^{del}_{Y_{i},t} +  q^{del}_{X_{f,i},t} p^{del}_{Y_{i},t} \big) + q^{del}_{X_{f,i},t} q^{del}_{Y,t} \Big) \bigg)\\
%											\prod_{j=1}^{n-r} \bigg( 1 - s \Big( h \big( p^{wt}_{X_{f,i},t} q^{wt}_{Y_{i},t} +  q^{wt}_{X_{f,i},t} p^{wt}_{Y_{i},t} \big) + q^{wt}_{X_{f,i},t} q^{wt}_{Y_{i},t} \Big) \bigg)
%											\end{array} \right] \numberthis
%\end{align*}










%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\section{Individual-Based Simulations} \label{IBM-App}
% \renewcommand{\theequation}{C\arabic{equation}}
% \setcounter{equation}{0}
% \renewcommand{\thefigure}{C\arabic{figure}}
% \setcounter{figure}{0}


%Unfortunately, estimating the fixation probability of new SLR-expanding inversions using Wright-Fisher simulations requires the rather strong assumption that deleterious allele frequencies in all locus $\times$ chromosome classes change deterministically while the inversion itself experiences genetic drift. To validate our Wright-Fisher simulation results, we also estimated fixation probabilities for SLR-expanding inversions using an individual-based model (IBM). We briefly summarize the structure of the model below.

%Our simulations modeled all $2N$ proto sex chromosomes in a diploid population of $N$ individuals with an even sex ratio ($3N/2$ proto-X chromosomes and $N/2$ proto-Y chromosomes), with binary allelic states corresponding to wild-type ($0$) and deleterious ($1$) alleles at $n_{tot} = 10^4$ selected sites. In each generation wild-type alleles at each locus mutate to deleterious variants at a rate $\mu = 2 \times 10^{-6}$. All chromosomes were initially populated with allelic states using pseudo-random binomial sampling assuming that all selected sites were initially at mutation-selection balance (i.e., $\hat{q}_i = \mu/hs$). Sex-specific individual fitness during the adult stage was calculated as the product of the genotypic fitness expressions across all $n = n_{tot}x$ loci spanned by the inversion. After mutation and selection, $N$ pairs of mothers and fathers were randomly sampled with replacement with probabilities weighted by their sex-specific relative fitness values. Prior to fertilization, recombination occurred between the maternally- and paternally derived X chromosomes in mothers, and between X chromosomes and Y chromosomes in fathers carrying a non-inverted Y, after which one recombinant X chromosome was randomly chosen as the maternally inherited X chromosome in the offspring. No recombination occurred between X chromosomes and inverted Y chromosomes. As in our Wright-Fisher simulations, we tracked the fate of single-copy inversion mutations until they either went extinct or fixed among Y chromosomes in the population. Prior to introducing the inversion, we ran the simulations without an inversion for $1000$ generations to allow allele frequencies to reach mutation-selection balance equilibrium. We performed $100*N$ replicate simulations to estimate the overall fixation probability for inversions of length $x$. Because these simulations were computationally intensive and took a long time, we limited our IBM simulations to inversions of length $x \leq 0.2$, which captured all inversion sizes with elevated fixation probabilities predicted by our W-F simulations, and used parameter values resulting in higher equilibrium frequencies of deleterious mutations ($h = 0.1$), which lessen the effect of drift on the frequency distribution of deleterious alleles.

%Computer code needed to reproduce the IBM simulations is available on GitHub at: \href{https://github.com/colin-olito}{https://github.com/colin-olito}.











%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Supplementary Figures} \label{SuppFigs-App}
 \renewcommand{\theequation}{C\arabic{equation}}
 \setcounter{equation}{0}
 \renewcommand{\thefigure}{S\arabic{figure}}
 \setcounter{figure}{0}


 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.55]{../../figures/deterministicSuppFig_h0_25}
 \caption{Overview of deterministic fitness and frequency dynamics for initially beneficial inversions of different sizes initially loaded with different numbers of deleterious alleles. Each column of panels presents results for inversions of lengths $x = 0.05$ (A--D), $0.1$ (E--H), $0.2$ (I--L), $0.5$ (M--P), and $0.8$ (Q--T). The first row of panels (A,E,I,M,Q) shows the probability that an inversion of length $x$ captures $r$ deleterious alleles (points and black lines), with benchmarks (vertical red dashed lines) showing the values of $r$ being illustrated in the corresponding column of panels. Values of $r$ were chosen to evenly cover the lower half of the distribution of $\Pr(r | x)$. As in Fig.~1 of the main text, the lower three rows of panels illustrate changes in inversion relative fitness ($2^{nd}$ row), inversion frequency ($3^{rd}$ row), and deleterious allele frequencies at {\itshape wt} loci on the inversion ($q_{Y_I}^{wt}$; red line), and both {\itshape wt} and {\itshape del} loci on X chromosomes in ovules/eggs ($q_{X_f}^{wt}$ and $q_{X_f}^{del}$; black solid and dashed lines respectively) ($4^{th}$ row). Results were generated using the following parameter values: $h = 0.2$, $s = 0.01$, $U = 0.02$, $n_{tot} = 10,000$.}
 \label{fig:DetermDynamics_h0.25}
 \end{figure}


 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.55]{../../figures/deterministicSuppFig_h0_1}
 \caption{Overview of deterministic fitness and frequency dynamics for initially beneficial inversions of different sizes initially loaded with different numbers of deleterious alleles. Each column of panels presents results for inversions of lengths $x = 0.05$ (A--D), $0.1$ (E--H), $0.2$ (I--L), $0.5$ (M--P), and $0.8$ (Q--T). The first row of panels (A,E,I,M,Q) shows the probability that an inversion of length $x$ captures $r$ deleterious alleles (points and black lines), with benchmarks (vertical red dashed lines) showing the values of $r$ being illustrated in the corresponding column of panels. Values of $r$ were chosen to evenly cover the lower half of the distribution of $\Pr(r | x)$. As in Fig.~1 of the main text, the lower three rows of panels illustrate changes in inversion relative fitness ($2^{nd}$ row), inversion frequency ($3^{rd}$ row), and deleterious allele frequencies at {\itshape wt} loci on the inversion ($q_{Y_I}^{wt}$; red line), and both {\itshape wt} and {\itshape del} loci on X chromosomes in ovules/eggs ($q_{X_f}^{wt}$ and $q_{X_f}^{del}$; black solid and dashed lines respectively) ($4^{th}$ row). Results were generated using the following parameter values: $h = 0.1$, $s = 0.01$, $U = 0.02$, $n_{tot} = 10,000$.}
 \label{fig:DetermDynamics_h0.1}
 \end{figure}



 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.55]{../../figures/deterministicSuppFig_h0_01}
 \caption{Overview of deterministic fitness and frequency dynamics for initially beneficial inversions of different sizes initially loaded with different numbers of strongly recessive deleterious alleles ($h = 0.01$). Each column of panels presents results for inversions of lengths $x = 0.05$ (A--D), $0.1$ (E--H), $0.2$ (I--L), $0.5$ (M--P), and $0.8$ (Q--T). The first row of panels (A,E,I,M,Q) shows the probability that an inversion of length $x$ captures $r$ deleterious alleles (points and black lines), with benchmarks (vertical red dashed lines) showing the values of $r$ being illustrated in the corresponding column of panels. Values of $r$ were chosen to evenly cover the lower half of the distribution of $\Pr(r | x)$. As in Fig.~1 of the main text, the lower three rows of panels illustrate changes in inversion relative fitness ($2^{nd}$ row), inversion frequency ($3^{rd}$ row), and deleterious allele frequencies at {\itshape wt} loci on the inversion ($q_{Y_I}^{wt}$; red line), and both {\itshape wt} and {\itshape del} loci on X chromosomes in ovules/eggs ($q_{X_f}^{wt}$ and $q_{X_f}^{del}$; black solid and dashed lines respectively) ($4^{th}$ row). Results were generated using the following parameter values: $h = 0.01$, $s = 0.01$, $U = 0.02$, $n_{tot} = 10,000$.}
 \label{fig:DetermDynamics_h0.01}
 \end{figure}



 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.75]{../../figures/PrFixSuppFix_h0_1}
 \caption{Fixation probabilities of new SLR-expanding inversions on Y chromosomes plotted as a function of inversion length when deleterious mutations are more strongly recessive ($h_i = 0.1$) than the average dominance coefficient of deleterious mutations estimated from empirical studies \citep{Manna2011, AgrawalWhitlock2012}. Point colors depict different population sizes ($N$), while point shapes indicate different chromosome-arm wide mutation rates relative to selection ($U/s$), which determines the average load of deleterious mutations per wild-type chromosome arm. Dashed horizontal lines indicate the expected fixation probability for a completely neutral variant for each value of $N$, and hence correspond to values of $2/N$ for Y-linked inversions. Other than the dominance coefficients, all parameter values were set to the same values as Fig.~2 in the main text: $h = 0.1$, $s = 0.01$, $n_{tot} = 10,000$.}
 \label{fig:PrFix_lessRecessive}
 \end{figure}


 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.75]{../../figures/PrFixSuppFix_h0_01}
 \caption{Fixation probabilities of new SLR-expanding inversions on Y chromosomes plotted as a function of inversion length when deleterious mutations are strongly recessive ($h_i = 0.01$). Point colors depict different population sizes ($N$), while point shapes indicate different chromosome-arm wide mutation rates relative to selection ($U/s$), which determines the average load of deleterious mutations per wild-type chromosome arm. Dashed horizontal lines indicate the expected fixation probability for a completely neutral variant for each value of $N$, and hence correspond to values of $2/N$ for Y-linked inversions. Other than the dominance coefficients, all parameter values were set to the same values as Fig.~2 in the main text: $h = 0.01$, $s = 0.01$, $n_{tot} = 10,000$. Note that when deleterious mutations are strongly recessive, there is a fixation bias towards larger inversions when the average mutation load per non-inverted chromosome arm is relatively low ({\itshape i.e.}, when $U/s = \{2, 5\}$), but this shifts to become a bias towards smaller inbersions at higher load levels (when $U/s = 10$). This contrasts qualitatively with our results for less recessive mutations presented in the main text (see Fix.~2). This qualitative change in model behaviour should be interpreted with caution, however, because the typical level of dominance for deleterious mutations across a chromosome arm is unlikely to be so strongly recessive ({\itshape e.g.}, \citealt{Manna2011, AgrawalWhitlock2012}). }
 \label{fig:PrFix_strongRecessive}
 \end{figure}



\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \section{Revisiting the conjecture of Charlesworth \& Wall (1999)} \label{CW1999-App}
 \renewcommand{\theequation}{D\arabic{equation}}
 \setcounter{equation}{0}
 \renewcommand{\thefigure}{D\arabic{figure}}
 \setcounter{figure}{0}

As explained in \ref{Sheltering-App}, the conjecture of \citet{CharlesworthWall1999} is based on the idea that under partial inbreeding, multiple selected loci under deleterious mutation pressure will behave similarly to a single selected locus under heterozygote advantage if they become linked to the SLR. Specifically, inbreeding is expected to generate associations between homozygosity and reduced fitness over a chromosome segment having multiple loci with segregating recessive deleterious variation \citep{Charlesworth1991b, Waller2021}. Below, we test this conjecture using a minimal example by extending the simulation model of \citet{CharlesworthWall1999} to accommodate three loci: a sex determining locus (SDL), and two selected loci under deleterious mutation pressure (locus $\mbf{A}$, with wild-type allele $A$ and deleterious variant $a$; and $\mbf{B}$, with corresponding alleles $B$, and $b$). In this relatively simple genetic system, the two selected loci should be able to generate the kind of multilocus apparent overdominance proposed by \citet{CharlesworthWall1999} when occuring in repulsion phase in double heterozygotes \citep{Charlesworth1991b, Waller2021}.


\subsection{Model} 

The three relevant loci are ordered $\mbf{SDL}$ - $\mbf{A}$ - $\mbf{B}$ on the proto sex chromosomes, and the SDL has two sex-determining "alleles", $X$ and $Y$, where $Y$ is the dominant male-determining factor. The wild-type alleles at $\mbf{A}$ and $\mbf{B}$ mutate to deleterious variants at a rate of $\mu$ and $v$ per meiosis respectively (with genotypic relative fitness expressions $w_{AA} = 1$, $w_{Aa} = 1 - h_1 s_1$, $w_{aa} = 1 - s_1$, and $w_{BB} = 1$, $w_{Bb} = 1 - h_2 s_2$, $w_{bb} = 1 - s_2$). The recombination rate between the SDL and $\mbf{A}$ is denoted $q$, and that between $\mbf{A}$ and $\mbf{B}$ is denoted $r$. The population is assumed to be large, and a fraction, $\alpha$, of all matings are between full siblings, while the remainder are random. Generations are discrete, and the order of life history events proceeds: (i) mutation, (ii) selection, (iii) meiosis, and (iv) mating. 

We track the fate of new inversion mutations  that link both selected loci to the SDL and capture a deleterious allele at one of the two selected loci (we arbitrarily assume the new inversion arises on a $Y_{Ab}$ haplotype, yielding an inverted $Y_{bA}^I$ haplotype). The conjecture requires that at least one deleterious allele becomes linked to the SDL, and since we also assume no back-mutation, we exclude the inverted $Y_{BA}^I$ genotype to reduce the size of the recursion system. Note that in the special case where $q = 1/2$, $r = 0$, $\mu = v = 0$, and $w_{ii}$ terms are given appropriate expressions for heterozygote advantage, the model is mathematically equivalent to the Y-autosome fusion model of \citet{CharlesworthWall1999}, but with two selected loci (Fig.~\ref{fig:3Loc-CW1999}).

\subsection{Analysis}

This genetic system can be described exactly by a system of $240$ recursions for the frequencies of matings between all possible pairs of genotypes ($X_{AB}/X_{AB} \times X_{AB}/Y_{AB}$, $X_{AB}/X_{AB} \times X_{AB}/Y_{Ab}$, etc.). The full system of recursions is provided in the accompanying R script in the Online Supplementary Material. Simulations were carried out in R \citep{RSoftware} for a variety of parameter conditions, but we focus our attention on scenarios of high and low recombination between the three loci, where deleterious mutations are completely or partially recessive ($h_1 = h_2 = \{0.0,\, 0.1\}$; $s_i = 0.01$). Under these conditions, mild deleterious mutations that are linked in repulsion (haplotypes involving $Ab$ or $aB$ allele pairings) should reduce the fitness of both segregating homozygotes, generating associative-overdominance (also called mutational heterosis) favoring heterozygosity at both loci \citep{Ohta1971a, Waller2021}.

Following \citet{CharlesworthWall1999}, we introduced a rare inversion haplotype at an initial frequency of $0.001$ into a population initially at mutation-selection balance, in males of genotype $X_{AB}/Y_{bA}^I$. Matings involving this genotype were assumed to be random in the first generation. We then iterated the system of recursions forward in time until the inversion either went extinct or reached a frequency of $0.01$. We then calculated the asymptotic rate of increase in log-frequency of the inversion to estimate the overall selection coefficient for the inversion when rare ($s_I$). 

\subsection{Simulation results}

As a first check that the 3-locus model is working correctly, we first reproduce the main result for Y-linked neo-sex chromosome fusions/translocations from \citet{CharlesworthWall1999}. As noted above (see \hyperref[Sheltering-App]{Appendix} \ref{Sheltering-App}), under the assumption that the selected loci initially recombine freely with the SDL prior to the inversion mutation, a model of an SLR-expanding inversion is equivalent to the models of \citet{CharlesworthWall1999} for neo-sex chromosome formation. We ran simulations for the same selection and dominance coefficients they used for heterozygote advantage at the $\mbf{A}$ locus, and set the relative fitness of all genotypes at the $\mbf{B}$ locus equal to $1$ (i.e., the $\mbf{B}$ locus is neutral). The simulations faithfully reproduced the main results of increasing invasion fitness with higher rates of partial full-sib mating (Fig.~\ref{fig:3Loc-CW1999}).

To explore the effect of linkage between the SDL and selected loci ($\mbf{A}$ and $\mbf{B}$) on the conjecture of \citet{CharlesworthWall1999}, we simulated the invasion of rare SDL-expanding inversions under four different linkage scenarios corresponding to a factorial cross between high and low levels of recombination between the SDL and $\mbf{A}$ locus and between the $\mbf{A}$ and $\mbf{B}$ locus ($q = \{0.5, 0.001\} \times r = \{0.5, 0.001\}$). Under the conjecture, linkage disequilibrium between the selected loci due to inbreeding and/or genetic linkage should promote associative overdominance, resulting in the pair of selected loci behaving similarly to a single locus under heterozygote advantage modelled by \citep{CharlesworthWall1999}. In this case, an inversion linking a deleterious allele at one locus and a wild-type allele at the other is hypothesized to be increasingly favoured in more inbred populations. However, our simulation results indicate that such an inversion is never selectively favoured under any of the linkage or dominance scenarios (Fig.~\ref{fig:3Loc-Sheltering-QR}; all selection coefficients for the inveresion are negative). In the most permissive scenario, when deleterious mutations are completely recessive ($h_i = 0$), an SLR-expanding inversion becomes nearly neutral in highly inbred populations.

 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.55]{../../figures/HetAdv3LocusCW1999}
 \caption{Reproduction of Fig.~1A from \citet{CharlesworthWall1999} using the 3-locus model of an SLR-expanding inversion. The figure shows the asymptotic rate of increase in log-frequency of the inversion when rare, which approximates the overall selection coefficient for the inversion, as a function of the frequency of full-sib matings ($\alpha$). Results are shown for several forms of heterozygote advantage: $s_1 = t_1 = 0.1$;  $s_1 = t_1 = 0.2$;  $s_1 = 0.1, t_1 = 0.2$;  $s_1 = 0.2, t_1 = 0.1$. Cases where there is no polymorphism at equilibrium prior to the inverseion are not shown.}
 \label{fig:3Loc-CW1999}
 \end{figure}


Taken in the broader context of our other model results presented in the main text, these supplementary simulation results suggest that the conjecture of \citet{CharlesworthWall1999} does not work for the following reason: despite potentially strong linkage disequilibrium between selected loci due to both inbreeding and genetic linkage -- which should favour associative overdominance at the selected loci -- recurrent deleterious mutations ensure that low-fitness homozygous genotypes are continually produced at the locus where the inversion captures a deleterious allele, much as they did in our models of randomly-mating populations. It therefore appears that additional conditions are necessary for this mechanism to favour the invasion of chromosomal inversions (or fusions or translocations between the Y chromosome and an autosome). For example, the models of \citet{deWaalMaleFijtCharlesworth1979}, in which autosomal translocation polymorphisms could be maintained under mutational heterosis, involved completely selfing hermaphrodite mating systems in which new deleterious mutations were quickly purged. 



 \begin{figure}[htbp]
 \centering
 \includegraphics[scale=0.625]{../../figures/sheltering3Locus_SuppFig}
 \caption{Effect of linkage between the SDL and selected loci ($\mbf{A}$ and $\mbf{B}$) on the invasion fitness of an SLR-expanding inversion capturing deleterious alleles in repulsion phase in the 3-locus model. The figure shows the asymptotic rate of increase in log-frequency of the inversion when rare, which approximates the overall selection coefficient for the inversion, as a function of the frequency of full-sib matings ($\alpha$). Results are shown for four different linkage scenarios corresponding to a factorial cross between high and low levels of recombination between the SDL and $\mbf{A}$ locus and between the $\mbf{A}$ and $\mbf{B}$ locus ($q = \{0.5, 0.001\} \times r = \{0.5, 0.001\}$) and two different dominance scenarios ($h_{i} = \{ 0.0, 0.1 \}$) for recessive deleterious mutations with a selection coefficient of $s_i = 0.1$ (where $i \in {\mbf{A}, \mbf{B}}$).}
 \label{fig:3Loc-Sheltering-QR}
 \end{figure}







\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \section{Revisiting the verbal hypothesis of Branco et al. (2018): Inbreeding and Linkage} \label{Branco-App}
 \renewcommand{\theequation}{E\arabic{equation}}
 \setcounter{equation}{0}
 \renewcommand{\thefigure}{E\arabic{figure}}
 \setcounter{figure}{0}

As highlighted in \hyperref[Sheltering-App]{Appendix} \ref{Sheltering-App}, the verbal 'sheltering' model proposed by \citet{Branco2017} suggests that prior linkage disequilibrium caused by a combination of inbreeding and partial linkage to mating-type or sex-determining genes makes selection against recessive deleterious alleles less efficient, thereby allowing deleterious alleles to increase in frequency at the margin of the non-recombining region. Rare recombination events are then suggested to generate individuals homozygous for (partially) recessive or deleterious alleles, which are selected against. Below, we develop a simple deterministic 2-locus model to study whether an inversion (or other large-effect recombination modifier) linking a deleterious allele at a \underline{single} selected locus to the dominant male-determining factor on the Y chromosome can invade a diploid population with X-Y genetic sex determination. Although this scenario is partially addressed by the previous 3-locus simulation model, the simpler 2-locus scenario is permissive of an analytic solution that offers additional insight.

\subsection{Model}
Consider a large population (i.e., genetic drift is negligible) of diploid, sexually reproducing individuals with discrete generations, in which sex is determined genetically by a dominant male-determining factor (i.e., a male heterogametic X-Y system). Our results are equally applicable to female heterogametic Z-W systems if male- and female-specific parameters are reversed. The population exhibits variation in the rate of inbreeding such that a fraction, $\alpha$, of the population has an inbreeding coefficient of $F$, while the remaining fraction $(1-\alpha)$ mates randomly. The gene(s) determining sex reside within a small non-recombining SLR, but recombination still occurs elsewhere along the chromosome at a rate $r$ per meiosis. As in our other models, we assume that genes located outside the SLR have functional homologs on both X and Y chromosomes. Generations are assumed to be discrete, and the order of life history events is: (i) mating, (ii) mutation, (iii) selection, and (iv) meiosis. 

We model the evolution of a large-effect recombination modifier (for simplicity we refer to the modifier as a chromosomal inversion) arising on a Y chromosome that (i) expands the MSY to include a single selected locus (the "load locus"); (ii) completely suppresses recombination between inverted and non-inverted karyotypes (in reality genetic exchange may rarely occur via double crossovers or gene conversion; \citealt{KrimbasPowell1992, KorunesNoor2019}); and (iii) has no direct fitness effects (i.e., has no breakpoint effects and causes no meiotic dysfunction). The wild-type allele ($A$) at the load locus mutates to a deleterious variant ($a$) at a rate $\mu$ per meiosis (with genotypic relative fitness expressions $w_{AA} = 1$, $w_{Aa} = 1 - h s$, $w_{aa} = 1 - s$), and is assumed to be under mutation-selection balance prior to the inversion mutation. Fixation of the inversion would lead to expansion of the non-recombining SLR, and fixation of the deleterious $a$ allele among Y chromosomes.

We assume the timescale for loss or fixation of the inversion is sufficiently short that dosage compensation is unlikely to evolve in the chromosomal region it spans. The critical question is whether there is any biologically plausible scenario in which a rare inversion capturing the deleterious variant can invade and fix among Y chromosomes. Below, we provide a brief analysis of this scenario under arbitrary levels of inbreeding and ancestral linkage between the SLR and load locus. A full derivation of the recursions and analysis of this model is provided in the accompanying Mathematica notebook file (.nb).

\subsection{Analysis}

The evolutionary dynamics of this genetic system can be represented by a system of four haplotype recursions describing the frequency changes of the deleterious $a$ allele at the load locus among the four relevant chromosome classes: X chromosomes among ovules/eggs ($X_f$), X chromosomes among pollen/sperm ($X_m$), non-inverted Y chromosomes ($Y$), and inverted Y chromosomes ($Y^I$). Under weak selection ($0 < s \ll 1$), the expected change in frequency of the inversion due to selection is expected to be slow relative to that due to inbreeding \citep{CaballeroHill1992, JordanConnallon2014, Olito2017}. In this case, it is reasonable to use a separation of timescales to approximate the per-generation change in haplotype frequencies \cite{OttoDay2007}. Specifically, we calculate the expected genotypic frequencies due to inbreeding in the absence of selection, and then substitute these quasi-equilibrium (QE) frequencies into the haplotype recursions before calculating the per-generation frequency change due to selection and mutation.

To identify parameter conditions under which the rare inversion haplotype is expected to invade, we evaluated the stability of the system of recursions for populations initially at mutation-selection balance (i.e., where $X_f=\hat{X}_f$, $X_m=\hat{X}_m$, $Y=\hat{Y}$, and $Y^I=0$). Under these assumptions, the overall selection coefficient for the inversion can be approximated by subtracting one from the eigenvalue of the Jacobian matrix associated with the change in frequency of inversion haplotypes ($s_I = \lambda_I - 1$). We are interested in identifying the parameter conditions under which the inversion is selectively favored (where $s_I > 0$). The conditions for the spread of the inversion will be most permissive when deleterious mutations are completely recessive ($h = 0$). In this case, the approximate selection coefficient for the inversion is

\begin{equation} \label{eq:sI_twoLoc}
	s_I = \frac{\alpha F (1 - X_f) + s(1 - Y) \big( \alpha F + 2 X_f(1 - \alpha F)(1 - \mu) + \mu(1 - \alpha F) \big)} {s \bigg(X_f \Big(\alpha F + 2 Y (1-\alpha F) \big(1 - \mu + \mu (1-\alpha F) \big) + Y \big(\mu+\alpha F(1-\mu) \big) \Big)-2 \bigg)}
\end{equation}
where $F$ is Wright's inbreeding coefficient, and $X_f$ and $Y$ are the equilibrium frequencies of the deleterious allele on X chromosomes in ovules/eggs and Y chromosomes prior to the inversion mutation. Evaluation of Eq(\ref{eq:sI_twoLoc}) reveals that it is negative for all biologically meaningful parameter space (i.e., for $0 \leq  \alpha,F,s,\mu,X_f,Y \leq 1$), indicating that no amount of inbreeding or prior linkage between the sex-determining and load loci can generate sufficient linkage disequilibrium to favor inversion establishment.






%%%%%%%%%%%%%%%%%%%%%
% Bibliography
%%%%%%%%%%%%%%%%%%%%%
\bibliography{./SuppInfo-bibliography}

\newpage

\end{appendices}

\end{document}
